docs/partial-inputs.ipynb

```diff
@@ -0,0 +1,446 @@
+    "        1597.68,  # green\n",
+    "        1741.10,  # blue\n",
+    "        2858.43,  # nir\n",
+    "    ]\n",
+    "    STD = [\n",
+    "        2026.96,  # red\n",
+    "        2011.88,  # green\n",
+    "        2146.35,  # blue\n",
+    "        2016.38,  # nir\n",
+    "    ]\n",
+    "\n",
+    "\n",
+    "data_dir = Path(DATA_DIR)\n",
+    "\n",
+    "dm = ClayDataModuleRGB(data_dir=str(data_dir.absolute()), batch_size=20)\n",
+    "dm.setup(stage=\"predict\")\n",
+    "trn_dl = iter(dm.predict_dataloader())"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "db3f3e5e-8668-4830-9c77-cc1d8cb35234",
+   "metadata": {},
+   "source": [
+    "### Create the embeddings for the images over the forest fire\n",
+    "This will loop through the images returned by the data loader\n",
+    "and evaluate the model for each one of the images. The raw\n",
+    "embeddings are reduced to mean values to simplify the data."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "id": "c5762240-9d22-4ebd-8e39-83fc6594a459",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Average embeddings have shape (12, 768)\n"
+     ]
+    }
+   ],
+   "source": [
+    "embeddings = []\n",
+    "\n",
+    "for batch in trn_dl:\n",
+    "    with torch.no_grad():\n",
+    "        # Move data from to the device of model\n",
+    "        batch[\"pixels\"] = batch[\"pixels\"].to(rgb_model.device)\n",
+    "        # Pass just the specific band through the model\n",
+    "        batch[\"timestep\"] = batch[\"timestep\"].to(rgb_model.device)\n",
+    "        batch[\"latlon\"] = batch[\"latlon\"].to(rgb_model.device)\n",
+    "\n",
+    "        # Pass pixels, latlon, timestep through the encoder to create encoded patches\n",
+    "        (\n",
+    "            unmasked_patches,\n",
+    "            unmasked_indices,\n",
+    "            masked_indices,\n",
+    "            masked_matrix,\n",
+    "        ) = rgb_model.model.encoder(batch)\n",
+    "\n",
+    "        embeddings.append(unmasked_patches.detach().cpu().numpy())\n",
+    "\n",
+    "embeddings = numpy.vstack(embeddings)\n",
+    "\n",
+    "embeddings_mean = embeddings[:, :-2, :].mean(axis=1)\n",
+    "\n",
+    "print(f\"Average embeddings have shape {embeddings_mean.shape}\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "72db5745-21c6-4b8e-b8f7-cb48c0f9c9ef",
+   "metadata": {},
+   "source": [
+    "## Analyze embeddings\n",
+    "Now we can make a simple analysis of the embeddings. We reduce all the\n",
+    "embeddings to a single number using Principle Component Analysis. Then\n",
+    "we can plot the principal components. The effect of the fire on the\n",
+    "embeddings is clearly visible.\n",
+    "\n",
+    "In the plot, each image embedding is one point. One can clearly distinguish\n",
+    "the two cloudy images and the values after the fire are consistently low.\n",
```

Comment:
Maybe good to move this sentence to a new markdown cell below the scatter plot, to be more like a caption describing the plot. I'd also recommend adding a sentence to remind the viewer to refer to the RGB plots above see the cloudy and burn scar images.

Reply:
good idea - done